### PR TITLE
arm64 support for Dev CiaB's prebuilt trafficserver image

### PR DESCRIPTION
**Pull request by zrhoffman**
This PR adds `linux/arm64` support to the `ghcr.io/apache/trafficcontrol/ci/trafficserver-alpine` image which makes the CDN in a Box for Developers `t3c` image work for users on `aarch64` machines, like M1 Macs.

This PR also makes it so that PRs and commits changing the `.env` file does not necessarily trigger rebuilding the `ghcr.io/apache/trafficcontrol/ci/trafficserver-alpine` image, it's only rebuilt when the `ATS_VERSION` line in the `.env` file changes (and when the workflow file or the container files are changed).
<hr/>

## Which Traffic Control components are affected by this PR?

- Dev CDN in a Box
- Automation - GitHub Container Registry

## What is the best way to verify this PR?

1. Push the [`zrhoffman:trafficserver-alpine-arm64`](https://github.com/zrhoffman/trafficcontrol/tree/trafficserver-alpine-arm64) branch to your own fork

2. Go to your fork -> Actions -> *Container ghcr.io/apache/trafficcontrol/ci/trafficserver-alpine* workflow -> Run workflow -> `trafficserver-alpine-arm64` branch and run the workflow

3. Wait 3ish hours for the workflow to complete

4. Go to https://ghcr.io/Your-Username/trafficcontrol/ci/trafficserver-alpine and verify that the ` ci/trafficserver-alpine` image was built for both the `linux/amd64` platform and the `linux/arm64` platform

5. Install an emulator for amd64 or arm64:
   ```shell
   docker run --privileged --rm tonistiigi/binfmt --install all
   ```
6. Verify that Traffic Server is in the amd64 image:
   ```shell
   docker pull --platform=linux/amd64 ghcr.io/Your-Username/trafficcontrol/ci/trafficserver-alpine
   docker run --rm ghcr.io/Your-Username/trafficcontrol/ci/trafficserver-alpine traffic_server --version
   ```
   Expected output (timestamps vary):
   ```trafficserver
   Traffic Server 9.1.2 Oct 31 2022 19:18:37 buildkitsandbox
   traffic_server: using root directory '/'
   Apache Traffic Server - traffic_server - 9.1.2 - (build # 103119 on Oct 31 2022 at 19:18:37)
   ```

7. Verify that Traffic Server is in the arm64 image:
   ```shell
   docker pull --platform=linux/arm64 ghcr.io/Your-Username/trafficcontrol/ci/trafficserver-alpine
   docker run --rm ghcr.io/Your-Username/trafficcontrol/ci/trafficserver-alpine traffic_server --version
   ```
   Expected output:
   ```trafficserver
   WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
   Traffic Server 9.1.2 Oct 31 2022 21:02:05 buildkitsandbox
   traffic_server: using root directory '/'
   Apache Traffic Server - traffic_server - 9.1.2 - (build # 103121 on Oct 31 2022 at 21:02:05)
   ```

When you are done testing, remove the emulators:
```shell
docker run --rm --privileged tonistiigi/binfmt --uninstall $(
  docker run --rm --privileged tonistiigi/binfmt --uninstall '' |
    jq -r '.supported + .emulators | join(",")'
)
```

## PR submission checklist
- [x] This PR has tests 
- [x] Added comments explaining unintuitive parts 
- [ ] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Review by ericholguin — APPROVED** [review]
LGTM. Images build properly and work for the right arch. Optionally `go mod vendor` could be added to `t3c/run.sh`

File: .github/containers/trafficserver-alpine/docker-compose.yml [hunk 1]
```diff
@@ -23,5 +23,22 @@ services:
       dockerfile: Dockerfile
       args:
         ATS_VERSION: ${ATS_VERSION}
+      # This docker-compose file is no longer used by the container-trafficserver-alpine.yml
+      # Github Actions workflow and is here
```
Comment (ocket8888) [inline .github/containers/trafficserver-alpine/docker-compose.yml:27]: > ... and is here

Seems like the sentence is missing its end?

**Review by ocket8888 — APPROVED** [review]

**Comment by ocket8888** [comment]
Assuming ericholguin has checked that this works on an arm processor (I don't have one that I wouldn't have to dig out of storage atm) then it's working fine.

File: .github/containers/trafficserver-alpine/docker-compose.yml (hunk 1 above)
Comment (zrhoffman) [reply .github/containers/trafficserver-alpine/docker-compose.yml]: Ended sentence in 42b3f57f63

**Pull request merged**